pkg/action/pull.go

```diff
@@ -125,7 +125,26 @@ func (p *Pull) Run(chartRef string) (string, error) {
 
 	saved, v, err := c.DownloadTo(chartRef, p.Version, dest)
 	if err != nil {
-		return out.String(), err
+		if registry.IsOCI(chartRef) && strings.Contains(fmt.Sprint(err), "remote error: tls: handshake failure") {
+			registryClient, err := registry.NewCrosClient(chartRef,
```

Comment:
Why are we re-creating a one-off registry client instead of altering `c.RegistryClient`? Shouldn't this setup code be part of [NewOciGetter](https://github.com/helm/helm/blob/c137bfb68b4f8b7a10687f043b0dbccb09574969/pkg/getter/ocigetter.go#L65)?

Reply:
Yes, you are right but there is one problem if we add that setup code in [NewOciGetter](https://github.com/helm/helm/blob/c137bfb68b4f8b7a10687f043b0dbccb09574969/pkg/getter/ocigetter.go#L65) then we need to distinguish between one-way TLS and two-way TLS. 
if we validate then a specific client setup will be done because we don't need a two-way tls client setup for one-way and vice-versa.

`if registry.IsOCI(chartRef) && strings.Contains(fmt.Sprint(err), "remote error: tls: handshake failure")`
so above code is validating when client couldn't establish connection(required client cert and key) then we setting up the client with roundtripper/tlsconfig data.

Reply:
**(When the Registry server is a one-way TLS that time it works fine  but when the registry server is a two-way TLS , the server looking for client certificate/key (two-way TLS), then the helm client does not provide the client info (TLSconfig) that causes a handshake failure. so my code is to resolve those issue)**

Reply:
Just make two-way TLS an `Option` like we do for the rest of the getter feature flags.

for example:

https://github.com/helm/helm/blob/c137bfb68b4f8b7a10687f043b0dbccb09574969/pkg/getter/getter.go#L54